File: Product of Array Except Self/Product_of_Array_Except_Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> prefix(n,0),suffix(n,0);
        int product = 1;
        for(int i =0;i<n; i++) {
            prefix[i] = product;
            product *=nums[i];
        }
        product = 1;
        for(int i =n-1; i>=0; i--) {
            suffix[i] = product;
            product *= nums[i];
        }
        vector<int> ans(n,0);
        for(int i =0; i<n; i++) {
            ans[i] = prefix[i]*suffix[i];
        }
        return ans;
    }
};

File: Product of Array Except Self/Product_of_Array_Except_Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        prefix = [1] * n
        suffix = [1] * n
        
        product = 1
        for i in range(n):
            prefix[i] = product
            product *= nums[i]
        
        product = 1
        for i in range(n-1, -1, -1):
            suffix[i] = product
            product *= nums[i]
        
        ans = [prefix[i] * suffix[i] for i in range(n)]
        return ans
        